proxy/http/HttpSM.cc

```diff
@@ -4242,6 +4244,12 @@ HttpSM::parse_range_and_compare(MIMEField *field, int64_t content_length)
       start = -1;
     } else {
       for (start = 0; s < e && *s >= '0' && *s <= '9'; ++s) {
+        // check the int64 overflow in case of high gcc with O3 option
+        // thinking the start is always positive
+        if (start >= cutoff && (start > cutoff || *s - '0' > cutlim)) {
```

Comment:
To the naked eye, this looks really weird. The whole checking of start twice, with slightly different range. But maybe this is needed as is ?

Reply:
I'm thinking the if conditional reads better with e.g.

```
(start > cutoff) || (start == cutoff && (*s -'0' > cutlim))
```

Reply:
And Maybe a comment like:

```
// Start > cutoff,  or when at cutoff and last digit > cutlim (8), we would overflow the additions to start
```


Or, after discussion with Bryan, maybe we should just use strtoll() here, which deals with overflow as well.